### PR TITLE
gpl: mbff updates from recent paper

### PR DESCRIPTION
**Pull request by therealshreyas**
Updates to the MBFF implementation from the recent paper:
(1) Remove constant knn value and use the Silhouette metric to find a good knn value.
(2) Fix AR-based distance bug.
(3) Updated default alpha, beta, and max_split_size default values to match experiments using ASAP7.
(4) Revert ILP to match implementation from old reference.
(5) Add the new paper reference.

### CODE REVIEW

File: src/stt/include/stt/SteinerTreeBuilder.h [hunk 1]
```diff
@@ -100,6 +100,12 @@ class SteinerTreeBuilder
                        const std::vector<int>& y,
                        const std::vector<int>& s,
                        int acc);
+
+  // run FOARS
+  Tree makeFoarsTree(const std::vector<int>& x_pin,
```
Comment (maliberty) [inline src/stt/include/stt/SteinerTreeBuilder.h:105]: Did you intend to include this in this PR?  It sems unrelated.

File: src/gpl/src/mbff.cpp [hunk 2]
```diff
@@ -137,6 +137,11 @@ float MBFF::GetDist(const Point& a, const Point& b)
   return (abs(a.x - b.x) + abs(a.y - b.y));
 }
 
+float MBFF::GetDistAR(const Point& a, const Point& b, const float AR)
```
Comment (maliberty) [inline src/gpl/src/mbff.cpp:140]: What is AR?

File: src/stt/include/stt/SteinerTreeBuilder.h (hunk 1 above)
Comment (therealshreyas) [reply src/stt/include/stt/SteinerTreeBuilder.h]: not sure why this happened again ... removed foars content in latest PR

File: src/gpl/src/mbff.cpp (hunk 2 above)
Comment (therealshreyas) [reply src/gpl/src/mbff.cpp:140]: aspect ratio-based distance for initial mbff generation

**Review by maliberty — APPROVED** [review]

**Pull request merged**